src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2927,7 +2963,7 @@ private bool HasUnboxingConversion(TypeSymbol source, TypeSymbol destination, re
             {
                 if (destination.IsValueType && !destination.IsNullableType())
                 {
-                    return true;
+                    return ConversionKind.ValueTypeUnboxing;
```

Comment:
this is interesting because destination could be a type parameter constrained on 'struct'.  As such, it coudl be seen as a type parameter unboxing, or a value type unboxing.

However, i'm taking the principle that we classify this according to the aspect of the check that the spec is referring to.  In this case, the spec is saying this converion is allowed if the destination is a value type.  Hence, it's thinking of it as a value-type unboxing, regardless of whether that value type is a type parameter or not.